Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkProducer.cc

```diff
@@ -1432,7 +1430,7 @@ std::pair<double, double> AlCaHcalIsotrkProducer::storeEnergy(const HcalRespCorr
 bool AlCaHcalIsotrkProducer::notaMuon(const reco::Track* pTrack0, const edm::Handle<reco::MuonCollection>& muonh) {
   bool flag(true);
   for (reco::MuonCollection::const_iterator recMuon = muonh->begin(); recMuon != muonh->end(); ++recMuon) {
-    if (recMuon->innerTrack().isNonnull()) {
+    if (recMuon->innerTrack().isNonnull() and recMuon->innerTrack().isAvailable()) {
```

Comment:
I'm wondering if this is the correct order for the condtions: if the inner track is not available, is the `isNonnul()` method returning anything reasonable?
Naively I would do:
```suggestion
    if (recMuon->innerTrack().isAvailable() and recMuon->innerTrack().isNonnull()) {
```

Reply:
in my naive not expert understanding:

- `isNonnull()`: checks whether the `edm::Ref` points to something (i.e. the reference is not null). It does not dereference the object, so it *should* be safe to call even if the object is unavailable.
- `isAvailable()`: checks whether the object being referenced is available in memory. This may involve dereferencing or accessing more internal state and could throw if the reference is null.

@cms-sw/core-l2 might comment as well if I understand this well.

Reply:
@mmusich's description is mostly correct. The `isAvailable()` is safe to call for a null `Ref` too, in which case it will return `false`.

The `isNonnull()` is also safe to call any time.

Reply:
Thanks for the input, then I guess I can simplify this check to just a check on `isAvailable()`